### PR TITLE
[syslog]: Add orchagent logrotate test

### PR DESCRIPTION
**Pull request by theasianpianist**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

#### How did you do it?
Use signals to pause/resume orchagent to reproduce the logrotate/pending APPL_DB entry issue

#### How did you verify/test it?
Tested with image 20201231.89 and verified that the test failed
Tested with image 20220531.38 and verified that the test passed

### CODE REVIEW

**Comment by prsunny** [comment]
@theasianpianist , can this be tested on the image with and without fix and provide the test results?

**Comment by theasianpianist** [comment]
> @theasianpianist , can this be tested on the image with and without fix and provide the test results?

I tested on both already, the test behaves as expected - fails on image without the fix, passes on image with the fix

File: tests/syslog/test_logrotate.py [hunk 1]
```diff
@@ -228,3 +231,57 @@ def test_logrotate_small_size(rand_selected_dut, simulate_small_var_log_partitio
...
+    # Unpause orchagent in case the test gets interrupted
+    rand_selected_dut.control_process('orchagent', pause=False)
+    clear_pending_entries(rand_selected_dut)
+
+
+# Sometimes other activity on the DUT can flush the missed notification during the test,
+# leading to a false positive pass. Repeat the test multiple times to make sure that it's
+# not a false positive
+@pytest.mark.repeat(5)
+def test_orchagent_logrotate(orch_logrotate_setup, rand_selected_dut):
```
Comment (prsunny) [inline tests/syslog/test_logrotate.py:277]: Can you point me to where logrotate is triggered?

Comment (theasianpianist) [reply tests/syslog/test_logrotate.py:277]: The log rotate is triggered by sending SIGHUP to orchagent, which configures a handler for this signal at init time.

**Review by prsunny — APPROVED** [review]

**Pull request merged**